src/dali/abstract_ccxt_input_plugin.py

```diff
@@ -544,7 +544,7 @@ def _process_buy(self, transaction: Any, notes: Optional[str] = None) -> Process
             in_transaction_list.append(
                 InTransaction(
                     plugin=self.plugin_name(),
-                    unique_id=transaction[_ID],
+                    unique_id=f"B{transaction[_ID]}",
```

Comment:
I was having issues with the transaction resolver linking the buy and sell transactions of JPY into transfers for some reason.

For Binance, the purchases are  by credit card. One Binance transaction becomes 3 in dali:

> 1) InTransaction for Deposit of JPY
> 2) OutTransaction for conversion of JPY (since I'm filing US taxes, this isn't a buy)
> 3) InTransaction of crypto asset

If I remember correctly, the transaction resolver combined 1) and 2) and interpreted it as a transfer.

The other case was I think with Bitbank, which PAYS maker fees (i.e. income for providing liquidity), a trade on that site also creates 3 transactions:

> a) OutTransaction for JPY purchases
> b) InTransaction of JPY income for providing liquidity
> c) InTransaction of crypto asset

In this case, the transaction resolver combined a) and b) and interpreted it as a transfer.

I guess I can just mangle the id for the Binance fiat deposit 1) and the id for the income from Bitbank b).

Reply:
This is making me realize that the transaction resolver should operate per-asset, so that no mangling would be necessary in a case like this, but instead it operates globally. So I would say: mangle the fiat transactions by appending `"/fiat_buy"` or `"/fiat_sell"`. Also maybe add comments describing these artificial fiat transactions.